### PR TITLE
Fix get_chunks to properly return surrounding chunks

### LINKED ISSUE #523: Chunk loader loads and unloads unnecessary chunks at the corners of surrounding area. Broken get_chunks
### How to reproduce

I. Unnecessary chunks

1. Set Game / Gameplay / View distance to 10000 & or disable fog
2. Load any world
3. Zoom out
Video 1

II. Broken `get_chunks`
1. Set Game / Gameplay / View distance to 10000 & or disable fog
2. Set Game / Gameplay / Chunk Load Distance to 6
3. Load any world
4. Zoom out
Image 2

### Observed behavior

Extra chunks loaded right now, later they will be unloaded, and the loaded once again

### Expected behavior

No extra chunks at the corners
All surrounding chunks properly loaded

### Version

- Game Version: master

### Additional information

Video 1
https://github.com/Libre-TrainSim/Libre-TrainSim/assets/935667/a565c61a-2702-45d3-b23c-75548a63a3db

Image 2
image

### PR DESCRIPTION
**Pull request by custodian**
Fixes #523

We can make code simple to understand with just iterating within `[-distance, distance+1]` as proposed within pr, as it is not necessary to optimize code right now.

or we can do only 1/4 or 1/8 semicircle rotations like
```
	var chunks := [
		chunk_to_string(Vector3(around.x, 0, around.z)),
	]
	for i in range(0, distance + 1):
		for j in range(0, distance + 1):
			if i + j == 0: continue
			var offset := Vector3(i, 0, j)
			if ceil(offset.length()) <= distance:
				var diagonal := Vector3(-i, 0, j)
				chunks.append_array([
					chunk_to_string(around + offset),
					chunk_to_string(around - offset),
					chunk_to_string(around + diagonal),
					chunk_to_string(around - diagonal),
					])
	return chunks
```

I vote for readability, so PR is plain and straight forward

### CODE REVIEW

**Comment by HaSa1002** [comment]
That's looking good. I just have to review it with a better tool than my phone xD

File: src/Data/Scripts/chunk_manager.gd [hunk 1]
```diff
@@ -150,10 +142,11 @@ func _unload_old_chunks(saving: bool = false):
 	var chunks_to_unload = loader._loaded_chunks.duplicate()
 
 	# chunks_to_unload = all chunks further away than treshold
+	var unload_distance = ProjectSettings["game/gameplay/chunk_unload_distance"]
```
Comment (HaSa1002) [inline src/Data/Scripts/chunk_manager.gd]:
```suggestion
	var unload_distance: float = ProjectSettings["game/gameplay/chunk_unload_distance"]
```
Please be explicit there

**Review by HaSa1002 — CHANGES_REQUESTED** [review]
verified in editor.
No cornered chunks appear.
Correct amount of chunks seem to load 👍 

Please fix the one style remark

File: src/Data/Scripts/chunk_manager.gd (hunk 1 above)
Comment (custodian) [reply src/Data/Scripts/chunk_manager.gd]: done, thanks

**Review by HaSa1002 — APPROVED** [review]
lgtm. @DasCapschen is gonna review later on

**Pull request merged**